actionmailbox/config/routes.rb

```diff
@@ -6,6 +6,12 @@
     post "/relay/inbound_emails"    => "relay/inbound_emails#create",    as: :rails_relay_inbound_emails
     post "/sendgrid/inbound_emails" => "sendgrid/inbound_emails#create", as: :rails_sendgrid_inbound_emails
 
+    # Amazon requires that SNS topic subscriptions have been accepted before sending notifications.
+    post "/amazon/inbound_emails"   => "amazon/confirmations#create", as: :rails_amazon_confirmations,
+         constraints: lambda { |request| request.params[:Type] == "SubscriptionConfirmation" }
```

Comment:
I don't think these constraints will work with real SNS. At least they aren't for me unless I have missed a config step.
SNS will set the Content-Type header to `text/plain` :( 
https://forums.aws.amazon.com/message.jspa?messageID=885234

So params is not going to be available because rails don't know to parse the json body.